### PR TITLE
updated readme and added wolf goat cabbage problem code

### PR DESCRIPTION
**Pull request by firekind**
Updated the readme title and added the code for the [wolf, goat, and cabbage problem](https://en.wikipedia.org/wiki/Wolf,_goat_and_cabbage_problem).<br>
The code is explained using comments.

### CODE REVIEW

**Review by mitrish — COMMENTED** [review]
Looks good

**Review by mitrish — APPROVED** [review]
Add a description of the problem statement as well. Add a link to the problem as well

File: graphs/wolf_goat_cabbage.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        Returns the initial state.
+        
+        Returns:
+            State: The initial state.
+        """
+
+        return State(left=['f', 'g', 'c', 'w'], right=[])
+
+
+if __name__ == "__main__":
```
Comment (rahulbansal16) [inline graphs/wolf_goat_cabbage.py]: @mitrish refactor the code in your PR, so that it can be run by anyone trying to copy the code

**Comment by rahulbansal16** [comment]
@firekind follow the pattern of @mitrish in writing the solution, Explain the problem and approach along with the possible optimizations and add the solution in that.

**Comment by firekind** [comment]
Added the explanation to the problem and approach. I also updated the code a bit.

**Pull request merged**